src/js/modules/infragistics.ui.widget.js

```diff
@@ -0,0 +1,137 @@
+/*!@license
+ * Infragistics.Web.ClientUI Base Widget <build_number>
+ *
+ * Copyright (c) 2011-<year> Infragistics Inc.
+ * <Licensing info>
+ *
+ * http://www.infragistics.com/
+ *
+ * Depends on:
+ *   jquery-1.9.1.js
+ *   jquery.ui.core.js
+ *   jquery.ui.widget.js
+ *   infragistics.util.js
+ *   infragistics.util.jquery.js
+ */
+
+(function (factory) {
+	if (typeof define === "function" && define.amd) {
+
+		// AMD. Register as an anonymous module.
+		define( [
+			"jquery",
+			"jquery-ui",
+			"./infragistics.util",
+			"./infragistics.util.jquery"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory(jQuery);
+	}
+}
+(function ($) {
+    /*
+		Base igWidget for all Ignite UI widgets.
+	*/
+    $.widget("ui.igWidget", {
+		localeWidgetName: null,
+        options: {
+            /* type="string|object" Set/Get the locale setting for the widget.
+            ```
+                 //Initialize
+                $(".selector").igWidget({
+                    locale: "ja"
+                });
+
+                // Get
+                var locale = $(".selector").igWidget("option", "locale");
+
+                // Set
+                $(".selector").igWidget("option", "locale", "ja");
+            ```
+            */
+            locale: "en",
+            /* type="string|object" Set/Get the regional setting for the widget.
+            ```
+                //Initialize
+                $(".selector").igWidget({
+                    regional: "ja"
+                });
+
+                // Get
+                var regional = $(".selector").igWidget("option", "regional");
+
+                // Set
+                $(".selector").igWidget("option", "regional", "ja");
+            ```
+            */
+			regional: "en-US"
+		},
+		_setOption: function (option, value) {
+			var prevValue = this.options[ option ];
+			if (prevValue === value) {
```

Comment:
As discussed, if locale and regional can be an object, getting the option value, changing some props and setting it back will not result in a change because of this line.